imports/methods/announcements/post.ts

```diff
@@ -0,0 +1,40 @@
+import { check } from 'meteor/check';
+import { Meteor } from 'meteor/meteor';
+import Ansible from '../../Ansible';
+import Announcements from '../../lib/models/Announcements';
+import MeteorUsers from '../../lib/models/MeteorUsers';
+import PendingAnnouncements from '../../lib/models/PendingAnnouncements';
+import { userMayAddAnnouncementToHunt } from '../../lib/permission_stubs';
+import TypedMethod from '../TypedMethod';
+
+export default new TypedMethod<{ huntId: string, message: string }, void>({
```

Comment:
Roughly speaking, if I don't pass the generic parameters explicitly, the compiler infers the type of `Args` and `Return` as their generic values (`Record<string, TypedMethodParam>` and `TypedMethodParam`) rather than inferring something more specific.

It seems like it should be possible to declare this in a way that the inference does work out, but I haven't gotten it yet. (I suspect it involves something like having a separate `Keys extends string` parameter with `validate` taking a `Record<Keys, unknown>` but I wasn't able to get that working either)

Reply:
Honestly, this feels mostly fine to me?  There's exactly one place that you are declaring what this method takes and what it returns and what its name is, which feels like keeping the API of a method all in one place.

I suppose it'd be neat if you could get everything to be inferred from the `check()` calls, but I'm not sad about having types for function signatures written down as clear API documentation, especially since we get the typechecker verifying (via our `check`s) that we're delivering on our claims.